src/main/java/seedu/address/logic/commands/EditListingCommand.java

```diff
@@ -0,0 +1,209 @@
+import static seedu.address.logic.parser.CliSyntax.PREFIX_REGION;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.listing.Address;
+import seedu.address.model.listing.Area;
+import seedu.address.model.listing.Listing;
+import seedu.address.model.listing.Price;
+import seedu.address.model.listing.Region;
+import seedu.address.model.person.Name;
+
+/**
+ * Edits the details of an existing listing in the system.
+ */
+public class EditListingCommand extends Command {
+
+    public static final String COMMAND_WORD = "editListing";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the listing identified "
+            + "by the index number used in the displayed listing list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_NAME + "NAME] "
+            + "[" + PREFIX_PRICE + "PRICE] "
+            + "[" + PREFIX_AREA + "AREA] "
+            + "[" + PREFIX_ADDRESS + "ADDRESS] "
+            + "[" + PREFIX_REGION + "REGION]...\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_PRICE + "4500 "
+            + PREFIX_AREA + "1200";
+
+    public static final String MESSAGE_EDIT_LISTING_SUCCESS = "Successfully edited listing: %1$s";
+    public static final String MESSAGE_DUPLICATE_LISTING = "This listing already exists in the system.";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_INVALID_LISTING_NAME = "The specified listing name is invalid.";
+
+    private final Name listingName;
+    private final EditListingDescriptor editListingDescriptor;
+
+    /**
+     * @param listingName of the listing in the filtered listing list to edit
+     * @param editListingDescriptor details to edit the listing with
+     */
+    public EditListingCommand(Name listingName, EditListingDescriptor editListingDescriptor) {
+        requireNonNull(listingName);
+        requireNonNull(editListingDescriptor);
+
+        this.listingName = listingName;
+        this.editListingDescriptor = new EditListingDescriptor(editListingDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Listing> lastShownList = model.getFilteredListingList();
+
+        Listing listingToEdit = lastShownList.stream()
+                .filter(listing -> listing.getName().equals(listingName))
+                .findFirst()
+                .orElse(null);
+
+        if (listingToEdit == null) {
+            throw new CommandException(MESSAGE_INVALID_LISTING_NAME);
```

Comment:
Similar to my previous comment, it may be better to avoid `null` checks and to use a contains() method instead to check whether the `listing` exists in the `model`.